### DISCUSSION TITLE
Data not sent every 15 mins on MQTT

### DISCUSSION THREAD

**Discussion by liams1989** [universam1/iSpindel#628] [Q&A]
Hi, I'm facing some strange issues where my data does not seem to be sending every 15 mins for example I've just checked and the battery was last updated 11 hrs ago, tilt & temp 3 hrs ago. I've got MQTT running in home assistant but not sure where to really start diagnosing what the issue is. Any thoughts?
![image](https://github.com/user-attachments/assets/c2ef776f-d7ab-4f10-a0c0-dde9919a6e61)
![image](https://github.com/user-attachments/assets/cdcec331-cb3e-46ad-a6a8-9cee2068b15a)

**Comment by thegreatgunbantoad** [comment]
I'd be inclined to try a free tier on brewfather (Simply as I know that to be pretty reliable and simple to set up do be aware of the need to end the device name [SG] if using SG rather than Plato) to see if it's the device itself, or the MQTT side that's dumping data.  How far is it from your router?  Is it inside of a metal fermenter/fridge?  It sounds like you're losing data rather than a fundamental setup problem, but i could be wrong.

**Comment by liams1989** [comment]
> I'd be inclined to try a free tier on brewfather (Simply as I know that to be pretty reliable and simple to set up do be aware of the need to end the device name [SG] if using SG rather than Plato) to see if it's the device itself, or the MQTT side that's dumping data.  How far is it from your router?  Is it inside of a metal fermenter/fridge?  It sounds like you're losing data rather than a fundamental setup problem, but i could be wrong.

I've got it right next to the router for testing. I've and I've been watching it and there is not even a blue led flash every 15 mins so wondering if there's more to it. I will try and connect to another service though just incase

**Reply by thegreatgunbantoad** [reply]
Is your wemos a genuine one or a replica?  I had troubles with some of the replicas before.  But for a while the genuine ones just couldn't be found

**Reply by thegreatgunbantoad** [reply]
Might be worth hooking up the UART to something like putty (Or you can do the whole platform IO thing but that's a little more involved).  Have you by any chance upgraded the firmware via it's web interface from a much older version? If so do a full reset.

**Reply by thegreatgunbantoad** [reply]
There was a change in file system somewhere and the upgrade across can get a little upset, a full upgrade wipes out the old settings file and rebuilds the file system.

**Reply by liams1989** [reply]
I've no idea if it's a genuine one or not as I purchased the unit pre built. It came with firmware 6.something from memory, tried doing an OTA upgrade from interface but it never worked so plugged it in via usb and updated firmware. Maybe I did something wrong there. Pretty sure I cleared it first but maybe not

**Reply by thegreatgunbantoad** [reply]
Do try a full reset from the AP.  If you can take a pic of the board and upload it, the not quite genuine ones are easy to spot if you know what to look for.

**Reply by thegreatgunbantoad** [reply]
There also might be the possibility that your server is throwing away data it thinks is garbage.  Is your polynomial setup correctly?  For SG a good starting point is: (1.0000E-6*(tilt^3))+(-1.3137E-4*(tilt^2))+(6.9680E-3*tilt)+(9.0704E-1)

**Reply by ErikdBr** [reply]
> I've no idea if it's a genuine one or not as I purchased the unit pre built. It came with firmware 6.something from memory, tried doing an OTA upgrade from interface but it never worked so plugged it in via usb and updated firmware. Maybe I did something wrong there. Pretty sure I cleared it first but maybe not

Did you switch the powerswitch to the off position during the firmware update? Otherwise you may have damaged the wemos if the diode wasn't removed from it.

See: https://github.com/universam1/iSpindel/issues/47

**Reply by thegreatgunbantoad** [reply]
> > I've no idea if it's a genuine one or not as I purchased the unit pre built. It came with firmware 6.something from memory, tried doing an OTA upgrade from interface but it never worked so plugged it in via usb and updated firmware. Maybe I did something wrong there. Pretty sure I cleared it first but maybe not
> 
> Did you switch the powerswitch to the off position during the firmware update? Otherwise you may have damaged the wemos if the diode wasn't removed from it.
> 
> See: #47

I've played fast and loose with that in the past and I don't think I ever killed anything.  I think I did do the maths at one point a came to the conclusion that there was not nearly enough current to cause harm.  That said that did not not consider the newer USB C / PD style ports, but since this doesn't support PD I still don't think there's a worry.

**Reply by liams1989** [reply]
> Do try a full reset from the AP.  If you can take a pic of the board and upload it, the not quite genuine ones are easy to spot if you know what to look for.

I'll get a picture tomorrow and do a full reset. It's strange because I'm sure you should see the LED flash when it sends the data? If so I'll change the range to a minute or so to see if that makes a difference for testing.

**Reply by thegreatgunbantoad** [reply]
Yes you should see an LED light up.  Your server will probabaly drop most of the data at that rate, but the LED should do it's thing regardless.

**Reply by liams1989** [reply]
So I've linked it to BrewSpy via https, had 3 reading every 15 mins then it had stopped again. 
I'm thinking of trying to clear the wemos completely then reload it to see if that resolves it. Otherwise I guess it's a new wemos

**Reply by thegreatgunbantoad** [reply]
Might be worth trying a different battery.  There's a lot of fake batteries out there.  It should weight in at ~ 42 g.

**Reply by liams1989** [reply]
So I've plugged it in via usb, and have consolled to it via putty. It seems to connect first time, however then I get this:
woken from deepsleep, normal mode
Samples:42 min:87.06 max:87.18 time:757
x: 15196 y: 6038 z: -808
Tilt: 87.12
Tacc: 21.83
Volt: 5.16
Temp: 20.56
Gravity: 1.11
After waiting 10207ms, result 7
Failed to connect -> trying to restore connection...
Rescued Wifi credentials
   -> waited for 11037ms, result 7
   -> Failed to restore connection...
                                     Final-sleep: 300s; RT: 22615

There seems to be some kind of issue reconnecting to the wifi from what I can gather, but I've no idea appart from that, or where to start with diagnosing it further. It is strange because it does connect sometimes, but then others clearly it doesn't. 

Has anyone else experienced this?

**Reply by ErikdBr** [reply]
Yes plenty of people, sounds like a resistor with the wrong value or e reversed diode or a faulty wemos

https://github.com/universam1/iSpindel/issues?q=is%3Aissue+resistor
